packages/fast-components-react-base/src/radio/radio.tsx

```diff
@@ -0,0 +1,124 @@
+}
+
+class Radio extends Foundation<IRadioHandledProps & IRadioManagedClasses, IRadioUnhandledProps, IRadioState> {
+
+    /**
+     * React life-cycle method
+     */
+    public static getDerivedStateFromProps(nextProps: IRadioHandledProps, prevState: IRadioState): null | Partial<IRadioState> {
+        if (typeof nextProps.checked === "boolean" && nextProps.checked !== prevState.checked) {
+            return {
+                checked: nextProps.checked
+            };
+        }
+
+        return null;
+    }
+
+    /**
+     * Handled props instantiation
+     */
+    protected handledProps: HandledProps<IRadioHandledProps & IManagedClasses<IRadioClassNameContract>> = {
+        checked: void 0,
+        disabled: void 0,
+        managedClasses: void 0,
+        onChange: void 0,
+        tag: void 0,
+        children: void 0
+    };
+
+    /**
+     * Provides reference to input
+     */
+    private inputRef: React.RefObject<HTMLInputElement>;
+
+    /**
+     * Define constructor
+     */
+    constructor(props: RadioProps) {
+        super(props);
+
+        this.state = {
+            checked: this.props.checked || false
+        };
+
+        this.inputRef = React.createRef();
+    }
+
+    public render(): React.ReactElement<HTMLElement> {
+        return (
+            <this.tag
+                {...this.unhandledProps()}
+                className={this.generateClassNames()}
+            >
+                <input
+                    className={get(this.props, "managedClasses.radio_input")}
+                    type="radio"
+                    ref={this.inputRef}
+                    onChange={this.handleRadioChange}
+                    disabled={this.props.disabled || null}
+                    checked={this.state.checked}
+                />
+                <span className={get(this.props, "managedClasses.radio_stateIndicator")} />
+                {this.renderedLabel()}
+            </this.tag>
+        );
+    }
+
+    /**
+     * Generates class names
+     */
+    protected generateClassNames(): string {
+        let classes: string = get(this.props, "managedClasses.radio");
+
+        classes = this.props.disabled ? `${classes} ${get(this.props, "managedClasses.radio__disabled")}` : classes;
+
+        return super.generateClassNames(classes);
+    }
+
+    /**
+     * Stores HTML tag for use in render
+     */
+    private get tag(): string {
+        return RadioTags[this.props.tag] || RadioTags.label;
+    }
+
+    private renderedLabel(): JSX.Element {
+        if (this.props.children) {
+            return(
+                <span className={get(this.props, "managedClasses.radio_label")}>
+                    {this.props.children}
+                </span>
+            );
+        }
+    }
+
+    /**
+     * Handles onChange as a controlled component
+     */
+    private handleRadioChange = (e: React.ChangeEvent<HTMLInputElement>): void => {
+        if (typeof this.props.checked !== "boolean") {
```

Comment:
I don't think that you actually need to manage state here in this component. You can use the "defaultChecked" property on the input to assign a checked value if no value prop is passed - that will handled the unhandled state. Then, if value is a boolean, it should be applied as the value of the input.

Reply:
add "slot"